src/linalg.rs

```diff
@@ -0,0 +1,696 @@
+use crate::error::Exception;
+use crate::utils::{IntoOption, MlxString, VectorArray};
+use crate::{Array, Stream, StreamOrDevice};
+use mlx_macros::default_device;
+use smallvec::SmallVec;
+use std::f64;
+
+#[derive(Debug, Clone, Copy)]
+pub enum Ord<'a> {
+    Str(&'a str),
+    P(f64),
+}
+
+impl<'a> Default for Ord<'a> {
+    fn default() -> Self {
+        Ord::Str("fro")
+    }
+}
+
+impl<'a> std::fmt::Display for Ord<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Ord::Str(s) => write!(f, "{}", s),
+            Ord::P(p) => write!(f, "{}", p),
+        }
+    }
+}
+
+impl<'a> From<&'a str> for Ord<'a> {
+    fn from(value: &'a str) -> Self {
+        Ord::Str(value)
+    }
+}
+
+impl<'a> From<f64> for Ord<'a> {
+    fn from(value: f64) -> Self {
+        Ord::P(value)
+    }
+}
+
+impl<'a> IntoOption<Ord<'a>> for &'a str {
+    fn into_option(self) -> Option<Ord<'a>> {
+        Some(Ord::Str(self))
+    }
+}
+
+impl<'a> IntoOption<Ord<'a>> for f64 {
+    fn into_option(self) -> Option<Ord<'a>> {
+        Some(Ord::P(self))
+    }
+}
+
+#[default_device]
+pub fn norm_p_device<'a>(
+    array: &Array,
+    ord: f64,
+    axes: impl Into<Option<&'a [i32]>>, // TODO: use IntoOption
```

Comment:
This could be resolved now, right?

Reply:
Resolved in [6202a12](https://github.com/oxideai/mlx-rs/pull/51/commits/6202a129a0267d7b45dcfd1b6fe6162934480eed)